hathor/p2p/sync_v2/manager.py

```diff
@@ -0,0 +1,1207 @@
+                return
+            data.append((h, blk_hash.hex()))
+        payload = json.dumps(data)
+        self.send_message(ProtocolMessages.PEER_BLOCK_HASHES, payload)
+
+    def handle_peer_block_hashes(self, payload: str) -> None:
+        data = json.loads(payload)
+        data = [(h, bytes.fromhex(block_hash)) for (h, block_hash) in data]
+        key = 'peer-block-hashes'
+        deferred = self.deferred_by_key.pop(key, None)
+        if deferred:
+            deferred.callback(data)
+
+    def send_get_next_blocks(self, start_hash: bytes, end_hash: bytes) -> None:
+        payload = json.dumps(dict(
+            start_hash=start_hash.hex(),
+            end_hash=end_hash.hex(),
+        ))
+        self.send_message(ProtocolMessages.GET_NEXT_BLOCKS, payload)
+        self.receiving_stream = True
+
+    def handle_get_next_blocks(self, payload: str) -> None:
+        self.log.debug('handle GET-NEXT-BLOCKS')
+        if self._is_streaming:
+            self.protocol.send_error_and_close_connection('GET-NEXT-BLOCKS received before previous one finished')
+            return
+        data = json.loads(payload)
+        self.send_next_blocks(
+            start_hash=bytes.fromhex(data['start_hash']),
+            end_hash=bytes.fromhex(data['end_hash']),
+        )
+
+    def send_next_blocks(self, start_hash: bytes, end_hash: bytes) -> None:
+        self.log.debug('start GET-NEXT-BLOCKS stream response')
+        try:
+            blk = self.tx_storage.get_transaction(start_hash)
+        except TransactionDoesNotExist:
+            # In case the tx does not exist we send a NOT-FOUND message
+            self.log.debug('requested start_hash not found', start_hash=start_hash.hex())
+            self.send_message(ProtocolMessages.NOT_FOUND, start_hash.hex())
+            return
+        assert isinstance(blk, Block)
+        assert blk.hash is not None
+        # XXX: it is not an error for the other peer to request a voided block, we'll pretend it doesn't exist, butf
+        blk_meta = blk.get_metadata()
+        if blk_meta.voided_by:
+            # In case the tx does not exist we send a NOT-FOUND message
+            self.log.debug('requested start_hash is voided, continue anyway', start_hash=start_hash.hex(),
+                           voided_by=[i.hex() for i in blk_meta.voided_by])
+            # XXX: we want to be able to not send this, but we do because the remote node could get stuck otherwise
+            # (tracked by issue #711)
+            # self.send_message(ProtocolMessages.NOT_FOUND, start_hash.hex())
+            # return
+        if self.blockchain_streaming is not None and self.blockchain_streaming.is_running:
+            self.blockchain_streaming.stop()
+        self.blockchain_streaming = BlockchainStreaming(self, blk, end_hash, limit=self.DEFAULT_STREAMING_LIMIT)
+        self.blockchain_streaming.start()
+
+    def send_get_prev_blocks(self, start_hash: bytes, end_hash: bytes) -> None:
```

Comment:
This is not used. We will use it only when the sync from checkpoints feature is implemented, right?

Reply:
That's right. I think we can remove it from this PR. Jan made a good point that keeping it would allow us to run the sync of checkpoint with old versions of hathor-core. But I think the implementation will likely change the code and it would be useless anyway.

Reply:
I thought I removed this already. Will remove now.